### PR TITLE
ref(ui): Refactor DiffModal to use GlobalModal

### PR DESCRIPTION
**Pull request by billyvg**
Refactor DiffModal to use GlobalModal

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/components/issueDiff.jsx [hunk 1]
```diff
@@ -113,9 +122,18 @@ const IssueDiff = createReactClass({
...
+  displayName: 'IssueDiffContainer',
+  mixins: [ApiMixin],
+  render() {
+    return <IssueDiff {...this.props} api={this.api} />;
   },
 });
 
-export default IssueDiff;
+export default IssueDiffContainer;
+export {IssueDiff};
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/issueDiff.jsx]: What do we export that for?

**Comment by lynnagara** [comment]
I think there's a bug in the snapshots

File: src/sentry/static/sentry/app/components/issueDiff.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/issueDiff.jsx]: For tests

**Comment by billyvg** [comment]
This modal compares 2 stacktraces, so what you're seeing is a stacktrace stub.

File: src/sentry/static/sentry/app/views/groupMerged/mergedToolbar.jsx [hunk 2]
```diff
@@ -59,7 +59,8 @@ const MergedToolbar = createReactClass({
 
     // only need eventId, not fingerprint
     let [baseEventId, targetEventId] = Array.from(entries).map(([, eventId]) => eventId);
-    ProjectActions.openDiffModal({
+
+    openDiffModal({
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/groupMerged/mergedToolbar.jsx]: Likely not related to your changes, but this modal doesn't work at all (nothing happens on click). Can you look into that?

**Review by lynnagara — APPROVED** [review]

File: src/sentry/static/sentry/app/components/modals/diffModal.jsx [hunk 3]
```diff
@@ -1,56 +1,25 @@
...
-  },
-
-  componentWillUnmount() {
-    if (this.unlisten) {
-      this.unlisten();
-    }
-  },
+class DiffModal extends React.Component {
+  static propTypes = {
+    Body: PropTypes.node,
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/modals/diffModal.jsx]: it's a function?

**Pull request merged**